magellan-legacy/src/main/java/com/wealthfront/magellan/Screen.java

```diff
@@ -199,26 +188,41 @@ protected void onRestore(Bundle savedInstanceState) {}
   protected void onUpdateMenu(Menu menu) {}
 
   /**
-   * Called when the Activity is resumed and when the Screen is shown.
+   * Called when the Screen in shown (including on rotation).
    */
-  protected void onResume(Context context) {}
+  @Override
+  @CallSuper
+  protected void onShow(@NotNull Context context) {
+    recreateView(context);
+  }
 
   /**
-   * Called when the Screen in shown (including on rotation).
+   * Called when the activity is resumed and when the Screen is shown.
    */
-  protected void onShow(Context context) {}
-
-  protected void onSave(Bundle outState) {}
+  @Override
+  protected void onResume(@NotNull Context context) { }
 
   /**
-   * Called when the Activity is paused and when the Screen is hidden.
+   * Called when the activity is paused and when the Screen is hidden.
    */
-  protected void onPause(Context context) {}
+  @Override
+  protected void onPause(@NotNull Context context) { }
 
   /**
    * Called when the Screen is hidden (including on rotation).
    */
-  protected void onHide(Context context) {}
+  @Override
+  @CallSuper
+  protected void onHide(@NotNull Context context) {
+    destroyView();
+  }
+
+  /**
+   * Called when the activity is killed.
+   */
+  @Override
+  @CallSuper
+  protected void onDestroy(@NotNull Context context) { }
```

Comment:
Do we need these empty override methods or can we just inherit them from `LifecycleAware`?

Reply:
We need these since its protected in the parent class.